services/galaxy-toolshed/galaxy-toolshed-activity.service.js

```diff
@@ -0,0 +1,70 @@
+import Joi from 'joi'
+import { formatDate } from '../text-formatters.js'
+import { age as ageColor } from '../color-formatters.js'
+import BaseGalaxyToolshedService from './galaxy-toolshed-base.js'
+
+const repositoryRevisionInstallInfoSchema = Joi.array().items(
+  Joi.object({
+    create_time: Joi.string().required(),
+  }).required(),
+  Joi.object({}),
+  Joi.object({})
+)
+
+export default class GalaxyToolshedActivity extends BaseGalaxyToolshedService {
+  static category = 'activity'
+  static route = {
+    base: 'galaxy-toolshed/create-date',
+    pattern: ':repositoryName/:owner',
+  }
+
+  static examples = [
+    {
+      title: 'Galaxy Toolshed - Create Time',
+      namedParams: {
+        repositoryName: 'sra_tools',
+        owner: 'iuc',
+      },
+      staticPreview: this.render({
+        createDate: new Date(0).setUTCSeconds(1538288239),
+      }),
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'create date',
+  }
+
+  static render({ createDate }) {
+    return { message: formatDate(createDate), color: ageColor(createDate) }
+  }
+
+  async fetch({ repositoryName, owner }) {
+    const changesetRevisions =
+      await this.fetchOrderedInstallableRevisionsSchema({
+        repositoryName,
+        owner,
+      })
+    return this.fetchRepositoryRevisionInstallInfoSchema({
+      schema: repositoryRevisionInstallInfoSchema,
+      repositoryName,
+      owner,
+      changesetRevision: changesetRevisions.shift(),
+    })
+  }
+
+  static transform(response) {
+    const metadata = response
+      .filter(function (x) {
+        return Object.keys(x).length > 0
+      })
+      .shift()
+    return metadata.create_time
+  }
```

Comment:
Perhaps there's a subtle distinction I've overlooked, but this entire block seems to be duplicated across all of these classes with the only differentiator being the key name used in the last line of the `transform` function. This seems like a prime candidate for simplification and consolidation by extracting and encapsulating into the base class or some other shared function definition with a signature that accepts any additional parameters (e.g. metadataFieldName) necessary to handle the minor differences

Reply:
You are right, I will update the code